packages/devtools_app/lib/src/timeline/flutter/timeline_controller.dart

```diff
@@ -29,81 +29,76 @@ const String timelineScreenId = 'timeline';
 class TimelineController implements DisposableController {
   TimelineController() {
     timelineService = TimelineService(this);
-    fullTimeline = FullTimeline(this);
-    frameBasedTimeline = FrameBasedTimeline(this);
-    timelines = [frameBasedTimeline, fullTimeline];
+    processor = TimelineProcessor(this);
   }
 
   final cpuProfilerController = CpuProfilerController();
 
   /// Notifies that a timeline event was selected.
-  ValueListenable get selectedTimelineEventNotifier =>
-      _selectedTimelineEventNotifier;
+  ValueListenable get selectedTimelineEvent => _selectedTimelineEventNotifier;
   final _selectedTimelineEventNotifier = ValueNotifier<TimelineEvent>(null);
 
+  /// Notifies that a timeline frame has been selected.
+  ValueListenable get selectedFrame => _selectedFrameNotifier;
+  final _selectedFrameNotifier = ValueNotifier<TimelineFrame>(null);
+
+  /// Notifies when an empty timeline recording finishes
+  ValueListenable get emptyRecording => _emptyRecordingNotifier;
+  final _emptyRecordingNotifier = ValueNotifier<bool>(false);
+
+  /// Notifies that the timeline is currently being recorded.
+  ValueListenable get recording => _recordingNotifier;
+  final _recordingNotifier = ValueNotifier<bool>(false);
+
+  /// Notifies that the recorded timeline data is currently being processed.
+  ValueListenable get processing => _processingNotifier;
+  final _processingNotifier = ValueNotifier<bool>(false);
+
+  /// Stream controller that notifies the timeline has been processed.
+  Stream<bool> get onTimelineProcessed => _timelineProcessedController.stream;
```

Comment:
Why is this one a Stream while the rest are ValueListenable?

Reply:
@terrylucas FYI